src/data.h

```diff
@@ -12,6 +12,13 @@
 #ifndef CYAML_DATA_H
 #define CYAML_DATA_H
 
+#include <assert.h>
+#if !defined(static_assert) && (defined(__GNUC__) || defined(__clang__)) \
+	&& defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L \
+	&& __STDC_VERSION__ <= 201710L
+#define static_assert _Static_assert
+#endif
```

Comment:
I've never let any of this sort of thing into the library before, because they inevitably grow as more people start building it for unusual platforms.

So I'd rather get to the bottom of why this toolchain is failing to build seemingly correct C11 source.

Failing that, my preference would be to revert a05e7db38073e5d1970e7797aafb46cfda668709.